### PR TITLE
Add trait Cpu and minor refactor

### PR DESCRIPTION
**Pull request by gallottino**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,3 +34,6 @@ target/
 
 # Ignore code-workspaces
 *.code-workspace
+
+# Test roms
+*.gba
```
Comment (guerinoni) [inline .gitignore:39]: No new line at the end :(

File: src/cpu.rs [hunk 2]
```diff
@@ -1,45 +1,20 @@
...
-
-    fn decode(&mut self, op: Vec<u8>) -> Condition {
-        let condition = op[3] & 0x0F; // latest 4 bit (28..32)
-        println!("condition -> {:b}", condition);
-        condition.into()
-    }
-}
+    // Abstraction of what happens for every instruction in the cpu
+    fn step(&self);
+}
```
Comment (guerinoni) [inline src/cpu.rs:20]: No new line at the end :(

File: src/condition.rs [hunk 3]
```diff
@@ -21,7 +21,7 @@ pub(crate) enum Condition {
 impl From<u8> for Condition {
     fn from(item: u8) -> Self {
         match item {
-            10 => Condition::GE,
+            0xA => Condition::GE,
```
Comment (guerinoni) [inline src/condition.rs:24]: Good!

**Review by guerinoni — CHANGES_REQUESTED** [review]
Only little things :) it also miss of `cargo fmt`

File: .gitignore (hunk 1 above)
Comment (gallottino) [reply .gitignore]: Done

File: src/cpu.rs (hunk 2 above)
Comment (gallottino) [reply src/cpu.rs]: Added

**Comment by guerinoni** [comment]
I will merge this, it should be nice for next PR rework same commit in order to avoid a commit that don't add nothing relevant, but just nitpicks :)

**Pull request merged**